include/z64animation.h

```diff
@@ -88,41 +69,185 @@ typedef struct {
+    /* 3 */ ANIMMODE_ONCE_INTERP,
+    /* 4 */ ANIMMODE_LOOP_PARTIAL,
+    /* 5 */ ANIMMODE_LOOP_PARTIAL_INTERP
+} AnimationMode;
+
+typedef enum {
+    /* -1 */ ANIMTAPER_DECEL = -1,
+    /*  0 */ ANIMTAPER_NONE,
+    /*  1 */ ANIMTAPER_ACCEL
+} AnimationTapers;
+
+#define ANIM_FLAG_UPDATEY (1 << 1)
+#define ANIM_FLAG_NOMOVE (1 << 4)
+
+typedef struct SkelAnime {
+    /* 0x00 */ u8 limbCount; // Number of limbs in the skeleton
+    /* 0x01 */ u8 mode; // See `AnimationMode`
+    /* 0x02 */ u8 dListCount; // Number of display lists in a flexible skeleton
+    /* 0x03 */ s8 taper; // Tapering to use when morphing between animations. Only used by Door_Warp1.
+    /* 0x04 */ void** skeleton; // An array of pointers to limbs. Can be StandardLimb, LodLimb, or SkinLimb.
+    /* 0x08 */ void* animation; // Can be an AnimationHeader or LinkAnimationHeader.
+    /* 0x0C */ f32 startFrame; // In mode ANIMMODE_LOOP_PARTIAL*, start of partial loop.
+    /* 0x10 */ f32 endFrame; // In mode ANIMMODE_ONCE*, Update returns true when curFrame is equal to this. In mode ANIMMODE_LOOP_PARTIAL*, end of partial loop.
+    /* 0x14 */ f32 animLength; // Total number of frames in the current animation.
+    /* 0x18 */ f32 curFrame; // Current frame in the animation
+    /* 0x1C */ f32 playSpeed; // Multiplied by R_UPDATE_RATE / 3 to get the animation's frame rate.
+    /* 0x20 */ Vec3s* jointTable; // Current translation of model and rotations of all limbs
+    /* 0x24 */ Vec3s* morphTable; // Table of values used to morph between animations
+    /* 0x28 */ f32 morphWeight; // Weight of the current animation morph as a fraction in [0,1]
+    /* 0x2C */ f32 morphRate; // Reciprocal of the number of frames in the morph
+    /* 0x30 */ union {
+        s32 (*normal)(struct SkelAnime*); // Can be Loop, Partial loop, Play once, Morph, or Tapered morph
+        s32 (*link)(struct PlayState*, struct SkelAnime*); // Can be Loop, Play once, or Morph
+    } update;
+    /* 0x34 */ s8 initFlags; // Flags used when initializing Link's skeleton
+    /* 0x35 */ u8 moveFlags; // Flags used for animations that move the actor in worldspace.
+    /* 0x36 */ s16 prevRot; // Previous rotation in worldspace.
+    /* 0x38 */ Vec3s prevTransl; // Previous modelspace translation.
+    /* 0x3E */ Vec3s baseTransl; // Base modelspace translation.
+} SkelAnime; // size = 0x44
+
+// Init
+
+BAD_RETURN(s32)
+SkelAnime_Init(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg,
+               AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
+BAD_RETURN(s32)
+SkelAnime_InitFlex(struct PlayState* play, SkelAnime* skelAnime, FlexSkeletonHeader* skeletonHeaderSeg,
+                   AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
+BAD_RETURN(s32)
+SkelAnime_InitSkin(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg,
+                   AnimationHeader* animation);
```

Comment:
not sure if this was on purpose or the formatter, but I think we should bring these back up on the same line as the type

Reply:
you mean `BAD_RETURN(s32)` on its own line? yeah that's the formatter
tbh I would prefer we deviate as little as possible from what the formatter does even in header files, it makes them much easier to maintain

Reply:
the formatter does the same in the .c btw
https://github.com/zeldaret/oot/blob/1587c5505ad66706ee53326c375c614c5c381e34/src/code/z_skelanime.c#L1438-L1440

Reply:
I dont really agree with that sentiment in general, since we exclude .h from formatting specifically so that we can have the freedom to do what we want. and I think it looks bad like this

Reply:
I won't say it looks good but at least it's consistent and easier to maintain than formatting by hand (unless we're okay with breaking line length limits on prototypes, then everything is a single line, however long, and we don't ever worry about it again)

Reply:
whenever I use the formatter on a header I go through and undo changes that look weird. its an understanding that youre not really supposed to use them there, but it can be convenient for some things

if you want to argue full committing to the formatter on everything, even headers, then its a discussion to have. but as things stand theres no need to be limited by it if we're not using it

Reply:
Should this be
```c
BAD_RETURN(s32) SkelAnime_Init(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg, AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
BAD_RETURN(s32) SkelAnime_InitFlex(struct PlayState* play, SkelAnime* skelAnime, FlexSkeletonHeader* skeletonHeaderSeg, AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
BAD_RETURN(s32) SkelAnime_InitSkin(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg, AnimationHeader* animation);
```

Reply:
id prob do something like this personally
```c
BAD_RETURN(s32) SkelAnime_Init(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg, 
                               AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
BAD_RETURN(s32) SkelAnime_InitFlex(struct PlayState* play, SkelAnime* skelAnime, FlexSkeletonHeader* skeletonHeaderSeg, 
                                   AnimationHeader* animation, Vec3s* jointTable, Vec3s* morphTable, s32 limbCount);
BAD_RETURN(s32) SkelAnime_InitSkin(struct PlayState* play, SkelAnime* skelAnime, SkeletonHeader* skeletonHeaderSeg, 
                                   AnimationHeader* animation);
```
which will prob look like shit on github, but its just roughly respecting our normal line limit with the macro still being on the same line (what youd expect the formatter to do in the first place if it wasnt dumb)

Reply:
alright, done